File: server/src/game/conf/PageConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PageConfigElem struct {
	Id int
	PageId int
	Bg int
	Type int
	Location []int
	Body int
	Scale float32
	Des string
	
}

type PageConfig0 struct {
	Dic map[int]*PageConfigElem
	Arr []*PageConfigElem
}

var (
	PageConfig PageConfig0
)

func loadPageConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PageConfig.Dic = make(map[int]*PageConfigElem)
	PageConfig.Arr = make([]*PageConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PageConfig.Arr)
	for _,conf := range PageConfig.Arr{
		PageConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/game/conf/PageConfig.ts
class PageConfig {
    private static _dic:Dictionary<PageConfig> = new Dictionary<PageConfig>();
    private static _arr:Array<PageConfig> = new Array<PageConfig>();

	 public Id :number ;
     public PageId :number ;
     public Bg :number ;
     public Type :number ;
     public Location : Array<number>  =new Array<number>();
     public Body :number ;
     public Scale :number ;
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PageConfig>{
        return  PageConfig._dic;
    }
    public static get Arr():Array<PageConfig>{
        return PageConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:PageConfig = <PageConfig>json[i];
				PageConfig._arr.push(config);
				PageConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("PageConfig loader fail!");
        }
    }
}